packages/redux-query-react/src/components/connect-request.js

```diff
@@ -54,6 +54,25 @@ const diffQueryConfigs = (
   return { cancelKeys, requestQueryConfigs };
 };
 
+export const headersChanged = (
+  queryConfigs: Array<QueryConfig>,
+  previousQueryConfigs: Array<QueryConfig>,
+) => {
+ return queryConfigs.map(function (config, i) {
+    if (previousQueryConfigs[i] && queryConfigs[i] && previousQueryConfigs[i].options && queryConfigs[i].options) {
+      var prevHeaders = previousQueryConfigs[i].options.headers;
+      var headers = queryConfigs[i].options.headers;
+
+      if (prevHeaders != null && headers != null) {
+        var prevHeaderValues = Object.values(prevHeaders);
+        return Object.values(headers).some(function (value, i) {
+          return prevHeaderValues[i] !== value;
```

Comment:
can it ever be `null === undefined`? but i think this is right

Reply:
I think this might be possible, updated and added a test case